app/src/androidTest/java/com/PreProcessorTest.java

```diff
@@ -112,4 +123,46 @@ public void performTest() {
             assertTrue(photoIsCorrect(images[i], faces[i]));
         }
     }
+
+    @Test
+    /**
+     * Testing if face on image works correctly.
+     */
+    public void IsFaceOnImageTest(){
+        Bitmap faceNotInImagePhoto = BitmapFactory.decodeResource(res, R.drawable.face_not_in_image);
+        Bitmap faceInImagePhoto = BitmapFactory.decodeResource(res, R.drawable.face_in_image);
+
+        Task<List<Face>> faceOutsideImage = preProcessor.detectAllFacesUsingML(faceNotInImagePhoto);
+        Task<List<Face>> faceInsideImage = preProcessor.detectAllFacesUsingML(faceInImagePhoto);
+
+        preProcessor.waitForTask(faceOutsideImage);
+        preProcessor.waitForTask(faceInsideImage);
+
+        assertFalse(preProcessor.isFaceOnImage(faceOutsideImage.getResult().get(0), faceNotInImagePhoto));
```

Comment:
assertTrue will fail if the second parameter evaluates to false (in other words, it ensures that the value is true). assertFalse does the opposite. This test checks nothing and always passes.

Reply:
i disagree